pcsx2/GS/Renderers/Common/GSRenderer.cpp

```diff
@@ -876,7 +876,31 @@ static std::string GSGetBaseFilename()
 std::string GSGetBaseSnapshotFilename()
 {
 	// prepend snapshots directory
-	return Path::Combine(EmuFolders::Snapshots, GSGetBaseFilename());
+	std::string base_path = EmuFolders::Snapshots;
+
+	// If organize by game is enabled, create a game-specific folder
+	if (GSConfig.OrganizeScreenshotsByGame)
+	{
+		std::string game_name;
+		if (std::string name = VMManager::GetTitle(true); !name.empty())
```

Comment:
same comment as the other function: seems to me you can avoid the extra `name` variable here.